src/ray/common/task/task_spec.h

```diff
@@ -29,11 +29,41 @@
 extern "C" {
 #include "ray/thirdparty/sha256.h"
 }
-
 namespace ray {
-typedef ResourceSet SchedulingClassDescriptor;
+/* typedef ResourceSet SchedulingClassDescriptor; */
 typedef int SchedulingClass;
 
+struct SchedulingClassDescriptor {
+ public:
+  explicit SchedulingClassDescriptor(ResourceSet &rs, FunctionDescriptor &fd, int64_t d)
```

Comment:
If there is a copy, usually we do either:

```
struct A {
A(const std::string& f): field(f) {}
std::string field;
}
```
or
```
struct A {
A(std::string f): field(std::move(f)) {}
std::string field;
}
```

(Prefer the later one since it gives the caller the ability to move).

So here, please add const to `fd` and `rs`.

Reply:
the second one looks better, so let's do that?